app/logic/Auth/PKCE.ts

```diff
@@ -0,0 +1,83 @@
+/* 
```

Comment:
Given that this is a third party library, either:
* Move this file to /lib/pkce/PKCE.ts
* Make a fork of the original library, and use the fork (I can publish the fork to npm, if you want)

Reply:
We could use the [pkce-challenge](https://www.npmjs.com/package/pkce-challenge) package as is. The only difference is that it returns an object with `code-verifier` and `code-challenge`together.
```
{
    code_verifier: 'u1ta-MQ0e7TcpHjgz33M2DcBnOQu~aMGxuiZt0QMD1C',
    code_challenge: 'CUZX5qE8Wvye6kS_SasIsa8MMxacJftmWdsIA_iKp3I'
}
```